test/e2e/multicast_test.go

```diff
@@ -0,0 +1,328 @@
+				group:           net.ParseIP("224.3.4.5"),
+			},
+			{
+				name:            "testMulticastForInterNodePods",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{1, false}},
+				port:            3457,
+				group:           net.ParseIP("224.3.4.6"),
+			},
+			{
+				name:            "testMulticastTrafficFromExternal",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: true},
+				receiverConfigs: []multicastTestPodConfig{{1, false}},
+				port:            3458,
+				group:           net.ParseIP("224.3.4.7"),
+			},
+			{
+				name:            "testMulticastTrafficToExternal",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{1, true}},
+				port:            3459,
+				group:           net.ParseIP("224.3.4.8"),
+			},
+		}
+		for _, mc := range testcases {
+			mc := mc
+			t.Run(mc.name, func(t *testing.T) {
+				runTestMulticastBetweenPods(t, data, mc)
+			})
+		}
+	})
+	t.Run("testMulticastBetweenPodsInThreeNodes", func(t *testing.T) {
+		skipIfNumNodesLessThan(t, 3)
+		testcases := []multicastTestcase{
+			{
+				name:            "testMulticastMultipleReceiversOnSameNode",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{0, false}, {0, false}},
+				port:            3460,
+				group:           net.ParseIP("224.3.4.9"),
+			},
+			{
+				name:            "testMulticastMultipleReceiversForInterNodePods",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{1, false}, {2, false}},
+				port:            3461,
+				group:           net.ParseIP("224.3.4.10"),
+			},
+			{
+				name:            "testMulticastMultipleReceiversTrafficFromExternal",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: true},
+				receiverConfigs: []multicastTestPodConfig{{1, false}, {2, true}},
+				port:            3462,
+				group:           net.ParseIP("224.3.4.11"),
+			},
+			{
+				name:            "testMulticastMultipleReceiversTrafficToExternal",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{1, true}, {2, false}},
+				port:            3463,
+				group:           net.ParseIP("224.3.4.12"),
+			},
+		}
+		for _, mc := range testcases {
+			mc := mc
+			t.Run(mc.name, func(t *testing.T) {
+				runTestMulticastBetweenPods(t, data, mc)
+			})
+		}
+	})
+	multipleInterfacesTested := false
+	for nodeIdx, ifaces := range nodeMulticastInterfaces {
+		if len(ifaces) >= 2 {
+			multipleInterfacesTested = true
+			t.Run("runTestMulticastForwardToMultipleInterfaces", func(t *testing.T) {
+				runTestMulticastForwardToMultipleInterfaces(t, data, nodeIdx, 3464, "224.3.4.13")
+			})
+			break
+		}
+	}
+	if !multipleInterfacesTested {
```

Comment:
```
t.Run("runTestMulticastForwardToMultipleInterfaces", func(t *testing.T) {
	multipleInterfacesFound := false
	var nodeIdx int
	for i, ifaces := range nodeMulticastInterfaces {
		if len(ifaces) >= 2 {
			multipleInterfacesFound = true
			nodeIdx = i
			break
		}
	}
	if !multipleInterfacesFound {
		t.Skip("Skipping test because none of the Nodes has more than one multicast enabled interface")
	}
	runTestMulticastForwardToMultipleInterfaces(t, data, nodeIdx, 3464, "224.3.4.13")
})
```
To leverage the `Skip` method so that the test will be executed anyway and marked as skipped like other skipped tests.

Reply:
updated